### PR TITLE
Telemetry continued

### PR DESCRIPTION
**Pull request by nico-bellante**

### CODE REVIEW

File: packages/editor/src/utils/typesafe-telemetry-actions.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+interface IMetadata {
+  telemetry?: ITelemetryMetadata;
+}
+
+interface ICreateActionOptions {
+  shouldSendTelemetry?: boolean;
+  addTimestamp?: boolean;
+}
+
+const defaultOptions = { shouldSendTelemetry: false, addTimestamp: false };
```
Comment (Zlatkovsky) [inline packages/editor/src/utils/typesafe-telemetry-actions.ts]: We can't be collecting telmetry until we go through a privacy review.  Could you add a constant here, like
```
// For now, until complete privacy review, only enabling telemetry in local dev environment, not production:
const TELEMETRY_ENABLED = process.env.NODE_ENV !== 'production'
```

... and then make use of it below?

File: packages/editor/src/pages/Editor/store/dialog/actions.ts [hunk 2]
```diff
@@ -13,18 +13,10 @@ interface IShowDialogProps {
...
-    subText,
-    buttons,
-    style = DialogType.largeHeader,
-    isBlocking = false,
-  }: IShowDialogProps) => resolve({ title, subText, buttons, style, isBlocking });
-});
+export const show = createAction('DIALOG_SHOW')<IShowDialogProps>();
 
-export const dismiss = createAction('DIALOG_DISMISS');
+export const dismiss = createAction('DIALOG_DISMISS')();
```
[the remark this answers is not in the document]
Comment (nico-bellante) [reply packages/editor/src/pages/Editor/store/dialog/actions.ts]: This is actually more inline with the original createAsyncAction API

File: packages/editor/src/utils/typesafe-telemetry-actions.ts (hunk 1 above)
Comment (nico-bellante) [reply packages/editor/src/utils/typesafe-telemetry-actions.ts]: @jakobpn

File: packages/editor/src/pages/Editor/store/dialog/actions.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { createAction } from 'typesafe-actions';
+import { createAction } from '../../../../utils/typesafe-telemetry-actions';
```
[the remark this answers is not in the document]
Comment (nico-bellante) [reply packages/editor/src/pages/Editor/store/dialog/actions.ts]: It wouldn't hurt to use the original typesafe-actions. The only time there would be a problem is if you wanted to add telemetry to those actions, which you could theoretically still do with the default typesafe actions, using this utility is just cleaner IMO

File: packages/editor/src/pages/Editor/store/samples/sagas.ts [hunk 4]
```diff
@@ -30,10 +29,6 @@ function* openSampleSaga(action: ActionType<typeof samples.get.request>) {
   const { content, error } = yield call(fetchYaml, url);
   if (content) {
     const solution = convertSnippetToSolution(content);
-    sendTelemetryEvent('Editor.SampleLoaded', [
-      oteljs.makeStringDataField('SampleName', solution.name),
```
Comment (AlexJerabek) [inline packages/editor/src/pages/Editor/store/samples/sagas.ts]: Are the sample name and ID still getting traced somewhere?

Comment (nico-bellante) [reply packages/editor/src/pages/Editor/store/samples/sagas.ts]: Yup, each action can define what to send from it's payload and type, you can see the sample's data here:
https://github.com/OfficeDev/script-lab/pull/707/files#diff-04980ac347c382f169d638e6d86d67daR16

File: packages/editor/src/utils/typesafe-telemetry-actions.ts (hunk 1 above)
Comment (Zlatkovsky) [reply packages/editor/src/utils/typesafe-telemetry-actions.ts]: Fine, I'll do it when it gets merged in.

**Comment by jakobpn** [comment]
Reviewed the latest commit and still need this change: Need to be able to specify event name for telemetry events.

**Pull request merged**